lua/ttt2/libraries/outline.lua

```diff
@@ -75,137 +216,237 @@ function outline.RenderedEntity()
     return RenderEnt
 end
 
-local function Render()
-    local client = LocalPlayer()
-    local IsLineOfSightClear = client.IsLineOfSightClear
-    local scene = render.GetRenderTarget()
-
-    render.CopyRenderTargetToTexture(storeTexture)
-
-    local w = ScrW()
-    local h = ScrH()
-
-    render.Clear(0, 0, 0, 0, true, true)
-
-    -- start stencil modification
-    render.SetStencilEnable(true)
-
-    cam.IgnoreZ(true)
-
-    render.SuppressEngineLighting(true)
-
-    render.SetStencilWriteMask(0xFF)
-    render.SetStencilTestMask(0xFF)
-
-    render.SetStencilCompareFunction(STENCIL_ALWAYS)
-    render.SetStencilFailOperation(STENCIL_KEEP)
-    render.SetStencilZFailOperation(STENCIL_REPLACE)
-    render.SetStencilPassOperation(STENCIL_REPLACE)
-
-    -- start cam modification
-    cam.Start3D()
-
-    for i = 1, ListSize do
-        local v = List[i]
-        local mode = v[MODE]
-        local ents = v[ENTS]
-
-        render.SetStencilReferenceValue(i)
-
-        for j = 1, #ents do
-            local ent = ents[j]
-
-            if
-                not IsValid(ent)
-                or mode == OUTLINE_MODE_NOTVISIBLE and IsLineOfSightClear(client, ent)
-                or mode == OUTLINE_MODE_VISIBLE and not IsLineOfSightClear(client, ent)
-            then
-                continue
-            end
+local function RenderModels(render_ents)
+    for j = 1, #render_ents do
+        local ent = render_ents[j]
 
+        if IsValid(ent) then
             RenderEnt = ent
-
-            ent:DrawModel()
+            ent:DrawModel(STUDIO_RENDER + STUDIO_NOSHADOWS)
         end
     end
+end
 
-    RenderEnt = NULL
-
-    cam.End3D()
-    -- end cam modification
-
-    render.SetStencilCompareFunction(STENCIL_EQUAL)
-
-    -- start cam modification
-    cam.Start2D()
-
-    for i = 1, ListSize do
-        render.SetStencilReferenceValue(i)
-
-        surface.SetDrawColor(List[i][COLOR])
-        surface.DrawRect(0, 0, w, h)
-    end
-
-    cam.End2D()
-    -- end cam modification
+local MATERIAL_WHITE = Material("models/debug/debugwhite")
+local mat_antialias = GetConVar("mat_antialias")
```

Comment:
is there a reason you chose two different naming styles? If you want to be in line with our style, it would be `materialDebugWhite` and `materialAntialias`

Reply:
the naming i used here mostly reflects how its done by valve in the source engine, in there convars are snakecased similar to how they are in the console because it both communicates that its a convar and what that convar is without having to look it up. Midway this PR ive changed my personal preferences for naming aswell which can be seen

Reply:
oh, the second one is no material, I'm blind :D

you can keep it that way or rename it to our style which is `cvEnableAntialias`